plutus-playground-client/test/Data/String/ExtraTests.purs

```diff
@@ -0,0 +1,26 @@
+module Data.String.ExtraTests
+       ( all
+       ) where
+
+import Prelude
+
+import Control.Monad.Eff.Random (RANDOM)
+import Data.String as String
+import Data.String.Extra (abbreviate)
+import Test.Unit (TestSuite, suite, test)
+import Test.Unit.QuickCheck (quickCheck)
+
+all :: forall eff. TestSuite (random :: RANDOM | eff)
+all =
+  suite "Data.String.Extra" do
+    abbreviateTests
+
+abbreviateTests :: forall eff. TestSuite (random :: RANDOM | eff)
+abbreviateTests = do
+  suite "abbreviate" do
+    test "Always limits the string length" do
+      quickCheck \str -> String.length (abbreviate str) <= 10
+    test "Never affects the start of the string" do
+      quickCheck \str ->
+        String.take 5 str ==
+        String.take 5 (abbreviate str)
```

Comment:
nice but even better would be to generate the length of the string, i.e. a string of max length 10. Use an arbitrary string and https://pursuit.purescript.org/packages/purescript-quickcheck/4.7.0/docs/Test.QuickCheck.Gen#v:chooseInt for example

```
do
  s <- arbitrary :: String
  l <- chooseInt 0 10
  pure $ take l s
```

Reply:
I don't see why that's better, in this case...?

Reply:
your algorithm could be broken for strings of length 7 and they are not tested in either test